apps/prairielearn/elements/pl-code/pl-code.py

```diff
@@ -112,6 +126,40 @@ def get_lexer_by_name(name: str) -> Optional[pygments.lexer.Lexer]:
             return None
 
 
+# Computing this is relatively expensive, so we'll do it once here and reuse it.
+@cache
+def get_ansi_color_tokens():
+    return color_tokens(ANSI_COLORS, ANSI_COLORS)
+
+
+# Instantiating a style/formatter is also expensive, so we'll cache it as well.
+# Note that sharing an instance means we'll also share the `hl_lines` instance
+# member, so the caller must take care to set or clear it as needed before use.
+@cache
+def get_formatter(
+    BaseStyle: type[pygments.style.Style], highlight_lines_color: Optional[str]
+):
```

Comment:
Nit: Is there no known type annotation for this and the above function?

Reply:
Are you talking about the type of the return value? Type inference is correctly doing its thing, but I'm happy to add an explicit annotation if you'd prefer.

Reply:
Generally prefer explicit annotations, the inference is usually very permissive.

Reply:
Explicit type annotation added in 4e4348f.